flink-end-to-end-tests/test-scripts/common_kubernetes.sh

```diff
@@ -48,10 +48,15 @@ function setup_kubernetes_for_linux {
       sudo rm "$(which minikube)"
     fi
 
+    # Retry loop download minikube.
     if ! [ -x "$(command -v minikube)" ]; then
       echo "Installing minikube $MINIKUBE_VERSION ..."
-      curl -Lo minikube https://storage.googleapis.com/minikube/releases/$MINIKUBE_VERSION/minikube-linux-$arch && \
-          chmod +x minikube && sudo mv minikube /usr/bin/minikube
+      download_minikube_url="https://storage.googleapis.com/minikube/releases/$MINIKUBE_VERSION/minikube-linux-$arch"
+      if ! retry_times ${MINIKUBE_START_RETRIES} ${MINIKUBE_START_BACKOFF} "curl -Lo minikube ${download_minikube_url}"; then
```

Comment:
have you tried the loop? Looks like curl doesn't return an error code in case of a failure. :thinking: hint: `--fail` might be what we want according to `man curl`